x-pack/legacy/plugins/reporting/export_types/common/execute_job/decrypt_job_headers.ts

```diff
@@ -25,9 +25,12 @@ export const decryptJobHeaders = async <
   job: JobDocPayloadType;
   logger: Logger;
 }): Promise<Record<string, string>> => {
-  const crypto: CryptoFactory = cryptoFactory(encryptionKey);
   try {
-    const decryptedHeaders: Record<string, string> = await crypto.decrypt(job.headers);
+    if (typeof job.headers !== 'string') {
+      throw new Error('Job headers are missing');
```

Comment:
`nodeCrypto` will throw an error when calling `Buffer.from` if it was not passed a string or a buffer https://github.com/elastic/node-crypto/blob/master/src/crypto.ts#L133

This will throw a more readable error and safe-guard the type

Reply:
Maybe this should be wrapped in `i18n.translate` ?

Reply:
I havent wrapped them in `i18n.translate` because this is the actual error payload and not the error message we are showing to the user.


The error message to the user is already i18n wrapped:
`'Failed to decrypt report job data. Please ensure that {encryptionKey} is set and re-generate this report. {err}'`.


I've wrapped them with `i18n.translation`; just explaining my thought process about that one.